agent/structs/config_entry_routes.go

```diff
@@ -103,10 +113,28 @@ func (e *TCPRouteConfigEntry) GetMeta() map[string]string {
 }
 
 func (e *TCPRouteConfigEntry) Normalize() error {
+	for i, parent := range e.Parents {
+		if parent.Kind == "" {
+			parent.Kind = APIGateway
+			e.Parents[i] = parent
+		}
+	}
 	return nil
 }
 
 func (e *TCPRouteConfigEntry) Validate() error {
+	validParentKinds := map[string]bool{
+		APIGateway: true,
+	}
+
+	if len(e.Services) > 1 {
+		return fmt.Errorf("tcp-based routes currently only support one service")
```

Comment:
I believe this would be clearer to just explicitly specify the type
```suggestion
		return fmt.Errorf("tcp-route currently only supports one service")
```